cmd/opampsupervisor/supervisor/config/config.go

```diff
@@ -40,5 +50,25 @@ type OpAMPServer struct {
 }
 
 type Agent struct {
-	Executable string
+	Executable  string
+	Description AgentDescription `mapstructure:"description"`
+}
+
+func (a *Agent) Validate() error {
+	return a.Description.Validate()
+}
+
+type AgentDescription struct {
+	IdentifyingAttributes    map[string]string `mapstructure:"identifying_attributes"`
+	NonIdentifyingAttributes map[string]string `mapstructure:"non_identifying_attributes"`
+}
+
+func (a *AgentDescription) Validate() error {
+	for k := range a.IdentifyingAttributes {
+		// Don't allow overriding the instance ID attribute
+		if k == semconv.AttributeServiceInstanceID {
+			return fmt.Errorf("cannot override identifying attribute %q", semconv.AttributeServiceInstanceID)
+		}
+	}
+	return nil
 }
```

Comment:
I'd like opinions on whether we should allow this to be overridden or not. The [spec states](https://github.com/open-telemetry/opamp-spec/blob/main/specification.md#agentdescriptionidentifying_attributes):
```
service.instance.id should be set. It may be be set equal to the Agent's instance uid (equal to ServerToAgent.instance_uid field) or any other value that uniquely identifies the Agent in combination with other attributes.
```

Which to me seems like it should be OK to override. However, the extension always sets `service.instance.id` to the opamp instance ID, even if the `service.instance.id` attribute on telemetry is different.

So I guess the question is, should we force `service.instance.id` to be consistent with opamp's instance_uid? Or should we remove this restriction?

Reply:
I think it should be OK to allow the end user to override `service.instance.id` if that's what they want and allow it to be different from opamp instance_uid. That's what the spec says.

However, I am not sure why exactly such override would be needed and what the use case is.

I am also open to the idea that we should not allow this if anyone has strong feelings about that. In that case we can also update the spec to make it a strict requirement for agent instance_uid and `service.instance.id` to be exactly the same instead of allowing implementations to decide.

There is also [PR in progress](https://github.com/open-telemetry/semantic-conventions/pull/950) to introduce `agent.id` and use that instead of `service.instance.id`. If that PR is accepted I think we have a stronger case to require it to always be equal to opamp instance_uid.

Reply:
I've removed the restriction for `service.instance.id`.